tower-layer/src/layer_fn.rs

```diff
@@ -0,0 +1,104 @@
+use super::Layer;
+use std::fmt;
+
+/// Returns a new `LayerFn` with the given closure.
+///
```

Comment:
Since this doc comment is the main documentation for `LayerFn` (the `LayerFn` type's docs just refer the reader to here), I think this should probably explain what `layer_fn` is *for*, not just what this function returns.

Maybe:
```suggestion
/// Returns a new [`LayerFn`] that implements [`Layer`] by calling the
/// given function.
///
/// The [`Layer::layer`] method takes a type implementing [`Service`] and
/// returns a different type implementing [`Service`]. In many cases, this can
/// be implemented by a function or a closure. The [`LayerFn`] helper allows
/// writing simple [`Layer`] implementations without needing the boilerplate of
/// a new struct implementing [`Layer`].
///
```
or something like that?

I realize this is basically copied from the docs for `service_fn`, which are basically identical...but I think the `service_fn` docs could use some improvement as well. :)